File: examples/v2/environments/go/main.go
package main

import (
	esc "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/esc"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		projectName := cfg.Get("projectName")
		if projectName == "" {
			projectName = "test-project"
		}
		envSuffix := cfg.Get("envSuffix")
		if envSuffix == "" {
			envSuffix = "dev"
		}

		environment, err := esc.NewEnvironment(ctx, "environment", &esc.EnvironmentArgs{
			OrgName: pulumi.String(serviceOrg),
			Project: pulumi.String(projectName),
			Name:    pulumi.String("testing-environment-" + envSuffix),
		})
		if err != nil {
			return err
		}

		ctx.Export("envName", environment.Name)
		return nil
	})
}

File: examples/v2/environments/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
organization_name = config.get("organizationName") or "service-provider-test-org"
project_name = config.get("projectName") or "test-project"
env_suffix = config.get("envSuffix") or "dev"

environment = ps_v2.esc.Environment(
    "environment",
    org_name=organization_name,
    project=project_name,
    name=f"testing-environment-{env_suffix}",
)

pulumi.export("environmentId", environment.id)
